components/automate-ui/src/app/pages/project/rules/project-rules.component.spec.ts

```diff
@@ -0,0 +1,219 @@
+        MockComponent({ selector: 'chef-table' }),
+        MockComponent({ selector: 'chef-thead' }),
+        MockComponent({ selector: 'chef-tbody' }),
+        MockComponent({ selector: 'chef-tr' }),
+        MockComponent({ selector: 'chef-th' }),
+        MockComponent({ selector: 'chef-td' }),
+        MockComponent({ selector: 'ng-container', inputs: ['hidden'] }),
+        ProjectRulesComponent
+      ],
+      imports: [
+        ReactiveFormsModule,
+        RouterTestingModule,
+        ChefPipesModule,
+        StoreModule.forRoot({
+          router: () => ({
+            state: {
+              url: '/projects/uuid-1/rules',
+              queryParams: {},
+              params: {},
+              fragment: '',
+              path: ['/']
+            },
+            previousRoute: {},
+            navigationId: 0
+          }),
+          projects: projectEntityReducer,
+          rules: ruleEntityReducer
+        })
+      ],
+      providers: [
+        FeatureFlagsService
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    jasmine.addMatchers(customMatchers);
+    fixture = TestBed.createComponent(ProjectRulesComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  describe('when creating a new rule', () => {
+    beforeEach(() => {
+      component.project = project;
+      component.rule = <Rule>{};
+      component.isLoading = false;
+      fixture.detectChanges();
+    });
+
+    it('the heading should show <Project Name>: Rule', () => {
+      const heading = `${component.project.name}: Rule`;
+      const compHeading = component.getHeading();
+      expect(heading).toBe(compHeading);
+    });
+
+    it('the heading should change to <Project Name>: <Rule Name>', () => {
+      const ruleName = 'My Rule';
+      component.ruleForm.get('name').setValue(ruleName);
+      const heading = `${component.project.name}: ${ruleName}`;
+      const compHeading = component.getHeading();
+      expect(heading).toBe(compHeading);
+    });
+
+    it('should have a backroute to project page', () => {
+      const backRoute = ['/settings', 'projects', component.project.id];
+      const compBackRoute = component.backRoute();
+      expect(backRoute).toEqual(compBackRoute);
+    });
+
+    it('the formgroup should have one condition', () => {
+      const conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(1);
+    });
+
+    it('should create a condition', () => {
+      component.addCondition();
+      const conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(2);
+    });
+
+    it('should delete a condition', () => {
+      let conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(1);
+      component.deleteCondition(0);
+      conditions = component.ruleForm.get('conditions').value.length;
+      expect(conditions).toBe(0);
```

Comment:
Technically this test is valid, but the user can never delete a condition when there is only one (as you have implemented). So two things to consider:
1. Add a couple more conditions then delete one, so it reflects what a user could actually do.
2. Disallow `deleteCondition()` from deleting the last condition. But then there's the issue of should we display an error in that case, and if so where... so maybe ignore this second point. 🤷‍♂